### PR TITLE
Emotes Menu

### PR DESCRIPTION
**Pull request by Morb0**
## About the PR

Adds emote menu that can be bind to key (default is `Y`) or opened from top menu bar.

**Icons for emotes not implemented in this PR, I have no such skill**

emotes2

## Why / Balance

Makes the use of emotions more accessible, understandable, consistent and comfortable.

## Technical details

- Uses `EntityWhitelist` for emote prototype whitelist and blacklist, to make emotes available only for certain entities (like borgs).
- Localize all emote names and messages
- Fix `EntityWhitelist` a `requireAll` check
- Remove unused `WaTail` emote

## Media

https://github.com/space-wizards/space-station-14/assets/14136326/bf51a81d-8f16-4394-bfc3-07513e8a2250

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added emote panel (by default Y)

### CODE REVIEW

**Comment by superjj18** [comment]
This is amazing, my only complaint is using L as the keybind as it requires me to lift either my WASD hand or my mouse hand, thus limiting robustness

**Comment by Morb0** [comment]
> This is amazing, my only complaint is using L as the keybind as it requires me to lift either my WASD hand or my mouse hand, thus limiting robustness

I took **L**iterally random button, so I open to suggestions

**Comment by Hmeister-real** [comment]
> > This is amazing, my only complaint is using L as the keybind as it requires me to lift either my WASD hand or my mouse hand, thus limiting robustness
> 
> I took **L**iterally random button, so I open to suggestions

As long as you can change the bind, this is amazing for new players, and people who don't like typing

**Comment by superjj18** [comment]
> > This is amazing, my only complaint is using L as the keybind as it requires me to lift either my WASD hand or my mouse hand, thus limiting robustness
> 
> I took **L**iterally random button, so I open to suggestions

I think Y might be available.  A bit of a left handed reach, but I think it would still work without needing to lift your hand from WASD too much

**Comment by FairlySadPanda** [comment]
Massive upvote for this feature, we were just talking about it in the voice chat yesterday.

**Comment by Hmeister-real** [comment]
> Would you consider adding #26594 to this feature?

they have, they show that they have in the video at the end

**Comment by IProduceWidgets** [comment]
The march towards controller support continues ever onwards

**Comment by trollbreeder** [comment]
Excellent work! This will make all sorts of emotes more accessible to players, as previously you had to learn from others or the code what emotes were what, but now it's all in one place and easily clickable.

I wonder if this could be made client configurable - then players could add their own emotes and categories to the wheel, similar to how emotes can be bound to keyboard with some config editing.

**Comment by SoulFN** [comment]
Its good, but needs more icons

**Comment by deltanedas** [comment]
> Icons for emotes not implemented in this PR, I have no such skill

be the change you want to see

**Comment by metalgearsloth** [comment]
Test fails real.

File: Resources/keybinds.yml [hunk 1]
```diff
@@ -187,6 +187,9 @@ binds:
 - function: OpenCharacterMenu
   type: State
   key: C
+- function: OpenEmotesMenu
+  type: State
+  key: L
```
Comment (mirrorcult) [inline Resources/keybinds.yml:192]:
```suggestion
- function: OpenEmotesMenu
  type: State
  key: Y
```

**Review by mirrorcult — APPROVED** [review]

**Comment by mirrorcult** [comment]
changing the default key to Y as proposed

**Pull request merged**